### PR TITLE
first version of the task service. 

### PR DESCRIPTION
**Pull request by jrantamaki**
Supports /task/ to get all tasks and /task/:id
- still missing the post and put http support

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -26,29 +26,33 @@ func main() {
 		c.File("static/index.html")
 	})
 
+	router.GET("/task/", func(c *gin.Context){
+		err, allTasks := service.GetAllTasks()
+		if err != nil {
+			c.JSON(http.StatusBadRequest, "Could not get all tasks");
```
Comment (Arch-vile) [inline main.go]: This is not bad reques, but server error?

File: main.go [hunk 2]
```diff
@@ -26,29 +26,33 @@ func main() {
...
-	log.Print("Hello Again and more. ",  anotherTestEntry.Description)
+	})
 
-	// example on how to serve json...
 	router.GET("/task/:id", func(c *gin.Context){
 		id :=  c.Param("id")
 		i, err := strconv.Atoi(id)
 
 		if err != nil { //FIXME there must be a better way to validate the parameter, that it is an int.
 			c.JSON(http.StatusBadRequest, "id must be a number");
```
Comment (Arch-vile) [inline main.go:45]: Could we have some helper method to handle getting required params with type?

File: model/task.go [hunk 3]
```diff
@@ -1,7 +1,12 @@
...
 
 type Task struct {
-	Description      string    `json:"description"`
-	Tags             []string  `json:"tags"`
-	Project			 string    `json:"project"`
+	Description string    `json:"description"`
+	Tags        []string  `json:"tags"`
 }
+// Tasks struct contains all tasks
+type Tasks struct {
```
Comment (Arch-vile) [inline model/task.go]: Not used?

**Review by Arch-vile — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (jrantamaki) [reply main.go]: yes, makes sense

File: main.go (hunk 2 above)
Comment (jrantamaki) [reply main.go:45]: yes. most likely there is a "by the book" way of doing it in gin... just have to figure out what it is.. :)

File: model/task.go (hunk 3 above)
Comment (jrantamaki) [reply model/task.go]: not yet... maybe I will remove it and bring it back when there is actually a  need for it.

**Pull request merged**